### ISSUE TITLE
JavaScript Error: "this.skipFields is undefined"

### ISSUE THREAD

**Issue by lucisandor** [ZotPlus/zotero-better-bibtex#175]
Version 0.8.17 exports one byte before crashing with "[JavaScript Error: "this.skipFields is undefined" {file: "cdsfvfdvfdvfd/translators/Better BibTeX.js" line: 975}]". This happens even when I give up on skipping fields.
Is there any way to get the previous version(s) or to downgrade?

**Comment by lucisandor** [comment]
Here's more:
```
(1)(+0000075): TypeError: this.skipFields is undefined
    ===== Stack Trace =====
    Translator.initialize/this.skipFields<@cdsfvfdvfdvfd/translators/Better BibTeX.js:975:5
    Translator.initialize@cdsfvfdvfdvfd/translators/Better BibTeX.js:973:22
    Translator.nextItem@cdsfvfdvfdvfd/translators/Better BibTeX.js:1102:3
    doExport@cdsfvfdvfdvfd/translators/Better BibTeX.js:8134:17
    =======================

(2)(+0000000): Translate: Translation using Better BibTeX failed: 
string => TypeError: this.skipFields is undefined
stack => Translator.initialize/this.skipFields<@cdsfvfdvfdvfd/translators/Better BibTeX.js:975:5
Translator.initialize@cdsfvfdvfdvfd/translators/Better BibTeX.js:973:22
Translator.nextItem@cdsfvfdvfdvfd/translators/Better BibTeX.js:1102:3
doExport@cdsfvfdvfdvfd/translators/Better BibTeX.js:8134:17
```

**Comment by AllThatIsTheCase** [comment]
An old file somehow made it into the release. I'm generating fresh and will republish when the tests have run

**Comment by AllThatIsTheCase** [comment]
A new release is out, and I'm looking into how this could have happened.

**Comment by AllThatIsTheCase** [comment]
I've changed the deployment process to always release binaries from a clean environment, and only after the tests have passed (yay CircleCI!)

**Comment by cazgp** [comment]
This error is still happening for me on the latest release v0.8.40.

**Comment by AllThatIsTheCase** [comment]
Are you sure it's exactly the same error? I've just downloaded 0.8.40 and unzipped it, and that code really isn't present anymore. Can you go into "Preferences->Advanced", check "enable after restart", then fully restart firefox (all windows must be closed), then re-open firefox, and select "report zotplus error" from the gear menu? If you click through you'll see a red URL, if you could post that here I'll have the details needed to look into this.

**Comment by cazgp** [comment]
Hey, so I only just saw this and fixed it already. I ended up deleting 
the add-on, purging the files from profile/translators, and 
re-installing the add-on. Not sure why the update didn't replace those 
files.

**Comment by AllThatIsTheCase** [comment]
That is strange. The plug-in actually replaces the translators on every
startup of zotero, regardless of whether they need updating or not. Perhaps
a file permission problem?

**Comment by lucisandor** [comment]
I had a slightly different error, but still localized to Better BibTeX.js, which won't go away, regardless of the version of ZBB I was using. Then I updated ZBB in both Zotero Standalone and Zotero addon to Firefox, and it went away. It could be that sharing a library between Standalone and Firefox messes translators.
I can't replicate any longer, but it may help cazgp.

**Comment by AllThatIsTheCase** [comment]
Profile sharing could very well mess things up; there is a known problem (#143) that makes things go screwy when both are running

**Comment by cazgp** [comment]
It definitely wasn't a file permission problem. Not quite sure what 
happened, but deleting and re-installing fixed it. Thanks!

**Comment by adriaanph** [comment]
i have been having a similar problem with symptoms:
1) exporting results in "An error occured while trying to export the selected file"
2) the log file contained "Zotero.BetterBibTeX is undefined" (could not import "chrome://zotero-better-bibtex/content/zotero-better-bibtex.js")
3) un-installing ZBB, purging translators, re-installing -- none of this worked. it seemed like ZBB didn't install correctly - i got the preferences panel but nothing was installed into the translators folder.

i am 100% certain the following fixed it:
4) updating zotero (from 4.0.21.2 to 4.0.26.4)
5) (re)installing ZBB (0.9.35)

**Comment by retorquere** [comment]
BBT will always only work with the latest released zotero. The interplay between zotero and BBT is too fine tuned. 

If you still experience problems, could you open a new issue?

**Comment by retorquere** [comment]
The error you mention seems to indicate an aborted install. Not seen that before, but such problems are usually fairly easy to diagnose - if this still happens (on the latest release), please do file a new issue, I'd love to get to the bottom of it.
